lib/chef/provider/service/macosx.rb

```diff
@@ -133,38 +149,49 @@ def disable_service
           unless @current_resource.enabled
             Chef::Log.debug("#{@new_resource} not enabled, not disabling")
           else
-            shell_out!(
-              "launchctl unload -w '#{@plist}'",
-              :user => @owner_uid, :group => @owner_gid
-            )
+            unload_service
+          end
+        end
+
+       def load_service
+          session = @session_type ? "-S #{@session_type} " : ''
+          cmd = 'launchctl load -w ' + session + @plist
+          shell_out_with_su?(cmd)
+        end
+
+        def unload_service
+          cmd = 'launchctl unload -w ' + @plist
+          shell_out_with_su?(cmd)
+        end
+
+        def shell_out_with_su?(cmd)
```

Comment:
Style question for the Rubyists - doesn't a method ending in `?` typically return a boolean?